tests/pick-test.c

```diff
@@ -12,9 +12,78 @@
 #include <string.h>
 #include <unistd.h>
 
+static void	 child(int, int);
+static void	 parent(int, int, const char *);
+static char	*parsekeys(const char *);
+static void	 sighandler(int);
+static void	 usage(void);
+
 static char	**pickargv;
 static int	  gotsig;
 
+int
+main(int argc, char *argv[])
+{
+	char	*keys = "";
+	pid_t	 pid;
+	int	 c, i, master, slave, status;
+
+	while ((c = getopt(argc, argv, "k:")) != -1)
+		switch (c) {
+		case 'k':
+			keys = parsekeys(optarg);
+			break;
+		default:
+			usage();
+		}
+	argc -= optind;
+	argv += optind;
+
+	/* Ensure room for program and null terminator. */
+	if ((pickargv = calloc(argc + 2, sizeof(const char **))) == NULL)
+		err(1, NULL);
+	pickargv[0] = "pick";
+	for (i = 0; i < argc; i++)
+		pickargv[i + 1] = argv[i];
+
+	if (signal(SIGCHLD, sighandler) == SIG_ERR)
+		err(1, "signal");
+	if ((master = posix_openpt(O_RDWR)) == -1)
+		err(1, "posix_openpt");
+	if (grantpt(master) == -1)
+		err(1, "grantpt");
+	if (unlockpt(master) == -1)
+		err(1, "unlockpt");
+	if ((slave = open(ptsname(master), O_RDWR)) == -1)
+		err(1, "%s", ptsname(master));
+
+	switch ((pid = fork())) {
+	case -1:
+		err(1, "fork");
+		/* NOTREACHED */
+	case 0:
+		child(master, slave);
+		/* NOTREACHED */
+	default:
+		parent(master, slave, keys);
+		/* Wait and exit with code of the child process. */
+		waitpid(pid, &status, 0);
+		if (WIFSIGNALED(status))
+			exit(128 + WTERMSIG(status));
+		if (WIFEXITED(status))
+			exit(WEXITSTATUS(status));
+	}
+
+	return 0;
+}
+
+static void
+usage(void)
+{
+	fprintf(stderr, "usage: pick-test [-k keys] [-- argument ...]\n");
+	exit(1);
+}
+
 static char *
```

Comment:
Correct. However, it's common to repeat the keyword at the definition as
well. I might revert the change to `src/pick.c` later.